smithy-cli/src/main/java/software/amazon/smithy/cli/dependencies/PinnedArtifact.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.cli.dependencies;
+
+import java.util.Objects;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+
+/**
+ * Pinned dependency definition used for locking dependencies.
+ */
+final class PinnedArtifact extends Artifact {
```

Comment:
Why is this is a subclass? It just has custom loading from a node, and it writes out the sum that the parent defines. Seems like it could be contained in the lock file implementation.

Reply:
I can change it to just be contained in the lockfile. I dont think the `Artifact` is necessary any more either.